packages/ui/src/composables/useValidation.ts

```diff
@@ -60,20 +64,37 @@ export const useValidation = (
       return true
     }
 
+    const promises: validationResult[] = []
     let error = false
     let errorMessages: string[] = []
 
     const rules = flatten(props.rules)
 
-    normalizeValidationRules(rules, props.modelValue)
-      .forEach((validationResult: boolean | string) => {
-        if (isString(validationResult)) {
-          errorMessages = [...errorMessages, validationResult]
-          error = true
-        } else if (validationResult === false) {
-          error = true
-        }
-      })
+    const validationResultHandler = (validationResult: validationResult) => {
+      if (isString(validationResult)) {
+        errorMessages = [...errorMessages, validationResult]
+        error = true
+      } else if (validationResult === false) {
+        error = true
+      } else if (validationResult instanceof Promise) {
+        innerLoadingState.value = true
+
+        promises.push(validationResult)
+      }
+    }
+
+    normalizeValidationRules(rules, props.modelValue).forEach(validationResultHandler)
+
+    if (promises.length) {
+      Promise.all(promises)
+        .then(promisedRules => promisedRules.forEach(validationResultHandler))
+        .catch(e => { throw new Error(e) })
```

Comment:
This catch looks redundant

Reply:
I wrote it here if the back-end validation returns error: network error, server error, etc.

Reply:
Yeah but then you throw that same error anyway. Essentially this catch replicates default behavior.